File: ParsFramework/Util/Main/Pars.Util.Puma/Pars.Util.Puma.Domain/LocaleData.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Runtime.Serialization;
using System.Text;
using System.Threading.Tasks;
using Pars.Core.Data;

namespace Pars.Util.Puma.Domain
{
    [DataContract]
    public class LocaleData : DomainBase<int>
    {
        [DataMember]
        public override int Ref => this.LocaleRef;

        [DataMember]
        public int LocaleRef { get; set; }

        private string _value;

        [DataMember]
        public string Value
        {
            get { return _value; }
            set
            {
                if (_value == value)
                    return;

                _value = value;
                HasChanged = true;
            }
        }
    }
}

File: ParsFramework/Util/Main/Pars.Util.Puma/Pars.Util.Puma.UI.Angular/src/app/definition/localedata.ts
export class LocaleData {
    LocaleRef: number;
    Value: string;
    HasChanged: boolean;

    constructor(localaRef: number, value: string, hasChanged: boolean) {
        this.LocaleRef = localaRef;
        this.Value = value;
        this.HasChanged = hasChanged;
    }

    public static createNew(): LocaleData {
        return new LocaleData(0, "", false);
    }
}